apps/arena/lib/arena/game/player.ex

```diff
@@ -213,6 +221,52 @@ defmodule Arena.Game.Player do
     |> round()
   end
 
+  def store_item(player, item) do
+    put_in(player, [:aditional_info, :inventory], item)
+  end
+
+  def inventory_full?(player) do
+    player.aditional_info.inventory != nil
+  end
+
+  def use_item(player, game_state) do
+    case player.aditional_info.inventory do
+      nil ->
+        game_state
+
+      item ->
+        player =
+          Enum.reduce(item.effects, player, &apply_effect/2)
+
+        put_in(player, [:aditional_info, :inventory], nil)
+
+        put_in(game_state, [:players, player.id], player)
+    end
+  end
+
+  def apply_effect({:recover_stamina, recover_stamina}, player) do
```

Comment:
I think this should be  `apply_effect(player, effect_to_apply)` so we can pipe the player to this function

Reply:
That would make sense, except this functions are used by the `Enum.reduce/3` in `user_item/2`, so the ordering is determined by that

Since this functions are exclusively used by that, I've marked as private instead of public and if the use case arises that we need them public we can refactor things to have them usable by both parts